File: SDKs/googleapis.com/networkservices/v1/go/pkg/models/shared/grpcroutefaultinjectionpolicydelay.go
package shared

// GrpcRouteFaultInjectionPolicyDelay
// Specification of how client requests are delayed as part of fault injection before being sent to a destination.
type GrpcRouteFaultInjectionPolicyDelay struct {
	FixedDelay *string `json:"fixedDelay,omitempty"`
	Percentage *int32  `json:"percentage,omitempty"`
}

File: SDKs/googleapis.com/networkservices/v1/typescript/src/sdk/models/shared/grpcroutefaultinjectionpolicydelay.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// GrpcRouteFaultInjectionPolicyDelay
/** 
 * Specification of how client requests are delayed as part of fault injection before being sent to a destination.
**/
export class GrpcRouteFaultInjectionPolicyDelay extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=fixedDelay" })
  fixedDelay?: string;

  @SpeakeasyMetadata({ data: "json, name=percentage" })
  percentage?: number;
}
